### PR TITLE
RONDB-412: RDRS Docker 

### PR DESCRIPTION
**Pull request by smkniazi**

### CODE REVIEW

File: recipes/rdrs.rb [hunk 1]
```diff
@@ -3,6 +3,18 @@
 conn_str_split = conn_str.split(/:/, 2)
 
 
+if node['ndb']['rdrs']['containerize'] == "true" 
+  bash 'Setting-RDRS-Image' do
+    user 'root'
```
Comment (kouzant) [inline recipes/rdrs.rb:8]: Use the `file_cache_path` as staging like we do at https://github.com/logicalclocks/ndb-chef/blob/master/recipes/install.rb#L179

File: templates/default/rdrs-start.sh.erb [hunk 2]
```diff
@@ -1,5 +1,20 @@
 #!/usr/bin/env sh 
 
+
+<% if node['ndb']['rdrs']['containerize'].eql? "true" -%>
+
+rm -f <%= node['ndb']['log_dir'] %>/rdrs.cid 
+docker run --rm  \
+  --name rdrs-<%= node['ndb']['version'] %>.service  \
```
Comment (kouzant) [inline templates/default/rdrs-start.sh.erb:8]: I believe the version in the container name is redundant

File: templates/default/rdrs-start.sh.erb [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+  --cidfile <%= node['ndb']['log_dir'] %>/rdrs.cid \
+  --mount type=bind,source=<%= node['ndb']['root_dir'] %>/rdrs_config.json,target=<%= node['ndb']['root_dir'] %>/rdrs_config.json,readonly  \
+  --mount type=bind,source=<%= @crypto_dir %>,target=<%= @crypto_dir %>,readonly  \
+  --network=host \
+  rdrs:<%= node['ndb']['version'] %> \
+  /srv/hops/mysql/bin/rdrs -config=<%= node['ndb']['root_dir'] %>/rdrs_config.json
```
Comment (kouzant) [inline templates/default/rdrs-start.sh.erb:14]: The pattern we usually follow is to mount the logs directory so the logs are available outside the container and are later picked up by filebeat. Here the logs are only accessible from the container.

Also, I think, due to the `--rm` flag if you stop the container, the logs are lost I guess

File: templates/default/rdrs-stop.sh.erb [hunk 4]
```diff
@@ -1,4 +1,9 @@
 #!/usr/bin/env sh 
+
+<% if node['ndb']['rdrs']['containerize'].eql? "true" -%>
+docker stop rdrs-<%= node['ndb']['version'] %>.service
```
Comment (kouzant) [inline templates/default/rdrs-stop.sh.erb:4]: Similarly, the version here is redundant

File: templates/default/rdrs.service.erb [hunk 5]
```diff
@@ -5,15 +5,26 @@ StartLimitIntervalSec=80
 StartLimitBurst=10
 
 [Service]
+
+<% if node['ndb']['rdrs']['containerize'].eql? "true" -%>
```
Comment (kouzant) [inline templates/default/rdrs.service.erb:9]: Suggestion: generally it's safer to use `casecmp?` instead of `eql?` as the former is case insensitive.

**Review by kouzant — CHANGES_REQUESTED** [review]

File: templates/default/rdrs-start.sh.erb (hunk 2 above)
Comment (smkniazi) [reply templates/default/rdrs-start.sh.erb:8]: there is no "belief" in engineering

File: recipes/rdrs.rb (hunk 1 above)
Comment (smkniazi) [reply recipes/rdrs.rb:8]: fixed

File: templates/default/rdrs-start.sh.erb (hunk 3 above)
Comment (smkniazi) [reply templates/default/rdrs-start.sh.erb:14]: fixed

**Review by smkniazi — COMMENTED** [review]
take a look

File: templates/default/rdrs-start.sh.erb (hunk 2 above)
Comment (kouzant) [reply templates/default/rdrs-start.sh.erb:8]: @smkniazi Let me put it in another way then :) I have never seen a systemd service or docker container where the name contains the version

File: recipes/rdrs.rb [hunk 6]
```diff
@@ -66,7 +67,7 @@
 end
 
 rdrs_log_file = "#{node['ndb']['rdrs']['log']['file_apth']}"
-if node['ndb']['rdrs']['log']['file_apth'] == "" 
+if node['ndb']['rdrs']['log']['file_apth'] == "" || node['ndb']['rdrs']['containerize'] == "true"
```
Comment (kouzant) [inline recipes/rdrs.rb:70]: There is a typo in `file_apth`

**Review by kouzant — CHANGES_REQUESTED** [review]

File: templates/default/rdrs-start.sh.erb (hunk 2 above)
Comment (SirOibaf) [reply templates/default/rdrs-start.sh.erb]: Yeah because we need to handle update and when we update we shutdown the previous version.

Comment (smkniazi) [reply templates/default/rdrs-start.sh.erb]: it is removed

**Review by kouzant — APPROVED** [review]

**Pull request merged**